plasmapy/particles/particle_class.py

```diff
@@ -439,37 +472,72 @@ def _validate_arguments(argument, Z, mass_numb):
         if Z is not None and not isinstance(Z, Integral):
             raise TypeError("Z is not an integer.")
 
-    def _initialize_attrs_categories(self):
-        """Create empty collections for attributes & categories."""
+    def _store_particle_identity(self):
+        """Store the particle's symbol and identifying information."""
+        argument, mass_numb, Z = self.__inputs
+        self._validate_arguments()
```

Comment:
```suggestion
        self._validate_arguments()

        argument, mass_numb, Z = self.__inputs
```

Just flipping it around so you're not unpacking `__inputs` twice before raising errors.